### PR TITLE
Odoo: Create tag for release numbers

### PR DESCRIPTION
**Pull request by LaurentGoderre**
/cc @d-fence

### CODE REVIEW

**Comment by LaurentGoderre** [comment]
Resolves #16618

**Comment by LaurentGoderre** [comment]
@whalelines how so?

**Comment by LaurentGoderre** [comment]
That I am not sure. I just augmented what was there

File: library/odoo [hunk 1]
```diff
@@ -2,15 +2,15 @@ Maintainers: Christophe Monniez <moc@odoo.com> (@d-fence)
 GitRepo: https://github.com/odoo/docker
 GitCommit: ea74918b211215176bda85854a771be1c5c97871
 
-Tags: 17.0, 17, latest
+Tags: 17.0, 17, 17.0-20240522, 17-20240522, latest
 Architectures: amd64, arm64v8, ppc64le
 Directory: 17.0
 
-Tags: 16.0, 16
+Tags: 16.0, 16, 16.0-20240522, 16-20240522
```
Comment (yosifkit) [inline library/odoo:9]: We don't rely on it often, but ordering the tags from most specific version to least specific version should minimize the diff here and in future PRs. We use the last tag in the list (the least specific) to help the diff minimize the "rename from -> rename to" output.
```suggestion
Tags:  16.0-20240522, 16-20240522, 16.0, 16
```

**Comment by d-fence** [comment]
hmm 🤔 will we have to accumulate the date tags in the file or just replace it ?

**Comment by LaurentGoderre** [comment]
@d-fence replace it.

**Comment by d-fence** [comment]
I'm ok with that then. Thanks for the PR

**Comment by yosifkit** [comment]
Yep, the old tags will stay on Docker Hub regardless of whether or not it is in the `library/odoo` file, so a new date tag each time is fine and expected with this change.

**Review by yosifkit — APPROVED** [review]

**Pull request merged**